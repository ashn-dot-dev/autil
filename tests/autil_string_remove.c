#define AUTIL_IMPLEMENTATION
#include "../autil.h"
#include "test.h"

int
main(void)
{
    struct autil_string* const s = autil_string_new_cstr("ABCDEF");

    autil_string_remove(s, 0, 0);
    ASSERT(strcmp(autil_string_start(s), "ABCDEF") == 0);

    autil_string_remove(s, 0, 1);
    ASSERT(strcmp(autil_string_start(s), "BCDEF") == 0);

    autil_string_remove(s, 2, 2);
    ASSERT(strcmp(autil_string_start(s), "BCF") == 0);

    // Remove bytes that slice out of bounds.
    autil_string_remove(s, 0, 4);
    ASSERT(strcmp(autil_string_start(s), "BCF") == 0);

    autil_string_del(s);
}
