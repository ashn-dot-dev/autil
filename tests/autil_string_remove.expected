error: [autil_string_remove] Invalid index,count 0,4
